app/Http/Controllers/FrontController.php

```diff
@@ -518,4 +516,29 @@ public function pilotGuideAtl() {
         return view('site.pilot_guide_atl')->with('controllers', $lcl_controllers)->with('controllers_update', $controllers_update)
         ->with('diag', $diag)->with('aaup', $aaup);
     }
+
+    private function reduceEventBanner(&$e) {
+        $disk = Storage::disk('public');
+        $filename = basename($e->banner_path);
+        $banner_path = 'event_banners/';
+        $reduced_path = $banner_path . 'reduced/';
+        if (!$disk->exists($reduced_path . $filename)) {
+            if ($disk->exists($banner_path . $filename)) {
+                $path = $disk->path($banner_path . basename($e->banner_path));
+                $directory = dirname($disk->path($banner_path . $filename));
+                list($width, $height) = getimagesize($path);
+                $new_width = 1500; // No reason for these banners to be > 1500 px wide
+                $new_height = ($new_width / $width) * $height;
+                $im = new \IMagick();
+                $im->readImage($path);
+                $im->resizeImage($new_width, $new_height, \Imagick::FILTER_LANCZOS, 0.9, true);
+                $im->writeImage($directory . '/reduced/' . $filename);
+                $im->destroy();
+            }
+        }
+        if ($disk->exists($reduced_path . $filename)) {
+            $directory = dirname($disk->path($banner_path . $filename));
+            $e->banner_path = $disk->url($reduced_path . $filename);
+        }
```

Comment:
If this logic is needed in 2 places, lets put this in the Event class so there is only one existence of the logic (so if we change something we only have to change it in one place.

Reply:
Concur - I'll do that